internal/services/mongocluster/mongo_cluster_resource.go

```diff
@@ -0,0 +1,572 @@
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-azurerm/helpers/azure"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+)
+
+type MongoClusterResource struct{}
+
+var _ sdk.ResourceWithUpdate = MongoClusterResource{}
+
+var _ sdk.ResourceWithCustomizeDiff = MongoClusterResource{}
+
+type MongoClusterResourceModel struct {
+	Name                       string            `tfschema:"name"`
+	ResourceGroupName          string            `tfschema:"resource_group_name"`
+	Location                   string            `tfschema:"location"`
+	AdministratorUserName      string            `tfschema:"administrator_username"`
+	AdministratorPassword      string            `tfschema:"administrator_password"`
+	CreateMode                 string            `tfschema:"create_mode"`
+	ShardCount                 int64             `tfschema:"shard_count"`
+	SourceLocation             string            `tfschema:"source_location"`
+	SourceServerId             string            `tfschema:"source_server_id"`
+	ComputeTier                string            `tfschema:"compute_tier"`
+	HighAvailabilityMode       string            `tfschema:"high_availability_mode"`
+	PublicNetworkAccessEnabled bool              `tfschema:"public_network_access_enabled"`
+	PreviewFeatures            []string          `tfschema:"preview_features"`
+	StorageSizeInGb            int64             `tfschema:"storage_size_in_gb"`
+	Tags                       map[string]string `tfschema:"tags"`
+	Version                    string            `tfschema:"version"`
+}
+
+func (r MongoClusterResource) ModelObject() interface{} {
+	return &MongoClusterResourceModel{}
+}
+
+func (r MongoClusterResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return mongoclusters.ValidateMongoClusterID
+}
+
+func (r MongoClusterResource) ResourceType() string {
+	return "azurerm_mongo_cluster"
+}
+
+func (r MongoClusterResource) Arguments() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"name": {
+			ForceNew: true,
+			Required: true,
+			Type:     schema.TypeString,
+			ValidateFunc: validation.StringMatch(
+				regexp.MustCompile(`^[a-z\d]([-a-z\d]{1,38}[a-z\d])$`),
+				"`name` must be between 3 and 40 characters. It can contain only lowercase letters, numbers, and hyphens (-). It must start and end with a lowercase letter or number.",
+			),
+		},
+
+		"resource_group_name": commonschema.ResourceGroupName(),
+
+		"location": commonschema.Location(),
+
+		"administrator_username": {
+			Type:         schema.TypeString,
+			Optional:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+			RequiredWith: []string{"administrator_username", "administrator_password"},
```

Comment:
We don't need to specify the property name that the `RequiredWith` is applied to
```suggestion
			RequiredWith: []string{"administrator_password"},
```

Reply:
Fixed.